frontend/views/containers/chatroom/file-attachment/ChatAttachmentPreview.vue

```diff
@@ -209,15 +230,34 @@ export default {
       if (attachment.url) {
         return attachment.url
       } else if (attachment.downloadData) {
-        const blob = await sbp('chelonia/fileDownload', new Secret(attachment.downloadData))
-        return URL.createObjectURL(blob)
+        const manifestCid = attachment.downloadData.manifestCid
+        const cachedArrayBuffer = await sbp('gi.db/filesCache/temporary/load', manifestCid)
+
+        if (cachedArrayBuffer) {
+          return URL.createObjectURL(new Blob([cachedArrayBuffer]))
+        } else {
+          const blob = await sbp('chelonia/fileDownload', new Secret(attachment.downloadData))
+          const arrayBuffer = await blob.arrayBuffer()
+          sbp('gi.db/filesCache/temporary/save', manifestCid, arrayBuffer)
```

Comment:
Initial implementation was to save a blob directly to the indexedDB which appeared to work well in the desktop browsers including Safari. But then I realised this does not work on iOS Safari. I referenced below existing code

https://github.com/okTurtles/group-income/blob/9e0241453fa40636f1e6b18aed51a517a2611fa6/frontend/views/components/Avatar.vue#L77-L79

Where a blob is converted to arrayBuffer before saving. Doing the same here fixed the issue.